packages/camera/camera_platform_interface/lib/src/types/camera_image.dart

```diff
@@ -0,0 +1,148 @@
+  bgra8888,
+}
+
+/// Describes how pixels are represented in an image.
+class ImageFormat {
+  ImageFormat._fromPlatformData(this.raw) : group = _asImageFormatGroup(raw);
+
+  /// Describes the format group the raw image format falls into.
+  final ImageFormatGroup group;
+
+  /// Raw version of the format from the Android or iOS platform.
+  ///
+  /// On Android, this is an `int` from class `android.graphics.ImageFormat`. See
+  /// https://developer.android.com/reference/android/graphics/ImageFormat
+  ///
+  /// On iOS, this is a `FourCharCode` constant from Pixel Format Identifiers.
+  /// See https://developer.apple.com/documentation/corevideo/1563591-pixel_format_identifiers?language=objc
+  final dynamic raw;
+}
+
+ImageFormatGroup _asImageFormatGroup(dynamic rawFormat) {
+  if (defaultTargetPlatform == TargetPlatform.android) {
+    // android.graphics.ImageFormat.YUV_420_888
+    if (rawFormat == 35) {
+      return ImageFormatGroup.yuv420;
+    }
+  }
+
+  if (defaultTargetPlatform == TargetPlatform.iOS) {
+    switch (rawFormat) {
+      // kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange
+      case 875704438:
+        return ImageFormatGroup.yuv420;
+      // kCVPixelFormatType_32BGRA
+      case 1111970369:
+        return ImageFormatGroup.bgra8888;
+    }
+  }
+
+  return ImageFormatGroup.unknown;
+}
+
+/// A single complete image buffer from the platform camera.
+///
+/// This class allows for direct application access to the pixel data of an
+/// Image through one or more [Uint8List]. Each buffer is encapsulated in a
+/// [Plane] that describes the layout of the pixel data in that plane. The
+/// [CameraImage] is not directly usable as a UI resource.
+///
+/// Although not all image formats are planar on iOS, we treat 1-dimensional
+/// images as single planar images.
+class CameraImage {
+  /// Creates a new [CameraImage] from the data received from the platform.
+  CameraImage.fromPlatformData(Map<dynamic, dynamic> data)
+      : format = ImageFormat._fromPlatformData(data['format']),
+        height = data['height'],
+        width = data['width'],
+        planes = List<Plane>.unmodifiable(data['planes']
+            .map((dynamic planeData) => Plane._fromPlatformData(planeData)));
+
+  /// Format of the image provided.
+  ///
+  /// Determines the number of planes needed to represent the image, and
+  /// the general layout of the pixel data in each [Uint8List].
+  final ImageFormat format;
+
+  /// Height of the image in pixels.
+  ///
+  /// For formats where some color channels are subsampled, this is the height
+  /// of the largest-resolution plane.
+  final int height;
+
+  /// Width of the image in pixels.
+  ///
+  /// For formats where some color channels are subsampled, this is the width
+  /// of the largest-resolution plane.
+  final int width;
+
+  /// The pixels planes for this image.
+  ///
+  /// The number of planes is determined by the format of the image.
+  final List<Plane> planes;
+}
```

Comment:
This class should be moved back to the `camera/camera` location. This class is only used when streaming image capture and will only be supported by Android and iOS.